packages/validator/src/slashingProtection/block/blockBySlotRepository.ts

```diff
@@ -68,7 +73,7 @@ export class BlockBySlotRepository {
   }
 
   async listPubkeys(): Promise<BLSPubkey[]> {
-    const keys = await this.db.keys(this.dbReqOpts);
+    const keys = await this.db.keys({gte: this.minKey, lt: this.maxKey, bucketId: this.bucketId});
```

Comment:
How is this not taken care of by the underlying db repository impl?

Reply:
The repositories on the validator do not extend [Repository](https://github.com/ChainSafe/lodestar/blob/3ed8d04c7971f753570f0d054a3c44bc2c382be1/packages/db/src/abstractRepository.ts#L19) and the [db itself](https://github.com/ChainSafe/lodestar/blob/3ed8d04c7971f753570f0d054a3c44bc2c382be1/packages/db/src/controller/level.ts#L143) does not handle that.

This is mostly do to the fact that there are incompatibilities with parameters and returns types which does not allow to override methods such as `encodeKey`.

Another more historic decision I found is that it was desired to keep db interface required by validator to be more minimal.
As mentioned in https://github.com/ChainSafe/lodestar/pull/1954 and a separate type [LodestarValidatorDatabaseController](https://github.com/ChainSafe/lodestar/blob/3ed8d04c7971f753570f0d054a3c44bc2c382be1/packages/validator/src/types.ts#L22).

Maybe a good refactor would be to create a base repository for the validator repositories with a minimal set of methods.

Reply:
I can look into creating a base repository for subset of methods required by validator repositories. Will have to see if that refactor even makes sense but would prefer to do it in another PR.

Reply:
I'll merge this for now, and feel free to open PR to dedup